### PR TITLE
Entity cannot be referenced validation failures for SMS message recipient_entity field

### PR DESCRIPTION
**Pull request by dpi**
- [Entity validation errors from DER base fields](https://www.drupal.org/node/2829049)
- [Entity cannot be referenced validation failures for SMS message recipient_entity field](https://www.drupal.org/node/2829041)

### CODE REVIEW

File: modules/sms_blast/src/Tests/SmsBlastWebTest.php [hunk 1]
```diff
@@ -60,16 +60,28 @@ protected function setUp() {
...
+    // Create users with multiple phone numbers. Only one message should be sent
+    // to each user.
+    $phone_numbers = $this->randomPhoneNumbers();
+    $entities = [];
+    for ($i = 0; $i < 6; $i++) {
+      /** @var \Drupal\user\UserInterface $user */
+      $user = $this->createEntityWithPhoneNumber($this->phoneNumberSettings, $phone_numbers);
+      // Need to activate so when DER does entity validation it is included by the
+      // UserSelection plugin.
+      $user->activate()->save();
```
Comment (almaudoh) [inline modules/sms_blast/src/Tests/SmsBlastWebTest.php:72]: Does this imply that every test that is written needs to activate the user before setting as sender or recipient entity?

File: src/Entity/SmsMessage.php [hunk 2]
```diff
@@ -444,7 +448,11 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
     $fields['recipient_entity'] = BaseFieldDefinition::create('dynamic_entity_reference')
       ->setLabel(t('Recipient entity'))
       ->setDescription(t('The entity who received the SMS message.'))
-      ->setRequired(FALSE);
+      ->setRequired(FALSE)
+      ->setSettings([
+        'exclude_entity_types' => TRUE,
```
Comment (almaudoh) [inline src/Entity/SmsMessage.php]: I've not gone through the code in detail, but it looks to me like DER already provides those defaults. This should imply a bug in DER.
```
class DynamicEntityReferenceItem extends EntityReferenceItem {

  /**
   * {@inheritdoc}
   */
  public static function defaultStorageSettings() {
    return [
      'exclude_entity_types' => TRUE,
      'entity_type_ids' => [],
    ];
  }
```

Comment (dpi) [reply src/Entity/SmsMessage.php]: Its weird, I cannot reproduce the fatal after re-downloading DER. Seems like co-incidence.

Looks like the settings are no longer required, so I have removed them from this PR. This PR now focuses on fixing the tests.

Posted an apology to DER https://www.drupal.org/node/2829049#comment-11793284

File: modules/sms_blast/src/Tests/SmsBlastWebTest.php (hunk 1 above)
Comment (dpi) [reply modules/sms_blast/src/Tests/SmsBlastWebTest.php:72]: The patch already addresses the few instances this is needed. You only need to activate the sender/recipient user if you are validating the sms message. Validation is currently only happening in `provider::queue` and `accountregistration` services.

Note: Validation does not occur as a part of saving entities, it must be done as an explicit step.

**Pull request closed without merging**